contracts/transfer/src/state.rs

```diff
@@ -20,23 +20,30 @@ use phoenix_core::{Crossover, Fee, Note, Ownable, StealthAddress};
 use poseidon_merkle::Opening as PoseidonOpening;
 use ringbuffer::{ConstGenericRingBuffer, RingBuffer};
 use rusk_abi::{
-    ContractError, ContractId, PaymentInfo, PublicInput, STAKE_CONTRACT,
+    ContractError, ContractId, EconomicMode, PaymentInfo, PublicInput,
+    RawResult, STAKE_CONTRACT,
+};
+use transfer_contract_types::{
+    EconomicEvent, EconomicResult, Mint, Stct, Wfct, Wfctc,
 };
-use transfer_contract_types::{Mint, Stct, Wfct, Wfctc};
 
 /// Arity of the transfer tree.
 pub const A: usize = 4;
 
 /// Number of roots stored
 pub const MAX_ROOTS: usize = 5000;
 
+// estimated cost of returning from the execute method
+const SURCHARGE_POINTS: u64 = 10000;
```

Comment:
There has been a lot of discussion in the past - and lots of changes in this repo to attest for them. We have never applied this strategy for a reason: there's no reason for it. It is not more costly when the transaction is takes up more computational resources, and it is not less costly when the transaction is more efficient.

It also seems to me that applying this strategy is totally orthogonal to this PR. We don't do this for "normal" transactions either. I think this should be removed, and the resulting code deleted as well.

Reply:
This also results in a pretty strange situation: what happens when the gas spent exceeds the gas limit due to this surcharge?

Reply:
It seems to me that this surcharge is meant to account for a part of the execution that is not actually accounted for when paying gas. 
If that's the case, I suggest to document (in a separate PR) what parts of the execution are actually subject to payment.
If already documented, a comment to the `execute` function mentioning this fact could help.

Reply:
I am fine with removing the surcharge - I added it for precision's sake. The idea was, we charge a bit more, and then the difference is just not paid by the user, so the overall balance of value is preserved. I am not sure we are talking about the same strategy, but for simplicity's sake, I will remove this surcharge. The downside will be that a tiny processor effort will be unaccounted for, but that is fine, as Federico said, there is still a lot of host code which is unaccounted for anyway, so no point of being so painstakingly precise here.